src/RequestRouter.m

```diff
@@ -64,9 +64,17 @@ - (void) registerRoute: (id<Route>) route {
 	NSArray *pathComponents = [self pathComponentsWithPath:path];	
 	__block NSObject<HTTPResponse> *response = nil;
 	for (id<Route> route in _routes) {
-        dispatch_sync(dispatch_get_main_queue(), ^{
+        /*
+         * If the first pathComponent is "direct_app_exec" then the execution should happen in THIS thread,
+         * not in the main UI thread as is true for all other commands.
+         */
+        if ([path hasSuffix:@"direct_app_exec"]) {
```

Comment:
I'm not really happy with this approach. Feels pretty hacky to hard-code intimate knowledge about that specific command in the path routing code.

Is there a way we can instead have the DirectAppExecCommand run its command in its own thread? That seems like it'd achieve the same goals.